app/models/blog.rb

```diff
@@ -10,20 +10,16 @@ class Blog < ApplicationRecord
   scope :published, -> { where('secret = FALSE') }
 
   scope :search, lambda { |term|
-    where('title LIKE ? OR content LIKE ?', "%#{term}%", "%#{term}%")
+    where('title LIKE :term OR content LIKE :term', { term: "%#{term}%" })
   }
 
   scope :default_order, -> { order(id: :desc) }
 
   scope :viewable, lambda { |user|
-    where('secret = false OR user_id = ?', user&.id)
+    where(secret: false).or(where(user_id: user&.id))
```

Comment:
```suggestion
    where(secret: false).or(where(user: user))
```

みたいにするとRailsがいい感じのSQLを作ってくれます。（ログを確認してみてください）

Reply:
同じSQLが作られているのが確認できました。
`user`を使うように修正します。